src/benchmarking/randomized_benchmarking.jl

```diff
@@ -0,0 +1,342 @@
+using LsqFit
+
+"""
+    RandomizedBenchmarkingFitProperties(model_order[, initial_parameters])
+
+Properties specifying how to generate a fit of the averaged sequence fidelity for a
+randomized benchmarking calculation.
+
+The models for the fit are described in [`RandomizedBenchmarkingFitResults`](@ref). The
+fit can have `model_order` 0, 1, or `nothing`. If `nothing` is specified, no fit will be
+generated. Optionally, a dictionary of initial fitting parameters can be passed to the
+constructor. The dictionary keys are symbols for the fitting parameters (e.g. "p").
+"""
+struct RandomizedBenchmarkingFitProperties
+    model_order::Union{Int, Nothing}
+    initial_parameters::Union{Dict, Nothing}
+
+    function RandomizedBenchmarkingFitProperties(model_order, initial_parameters)
```

Comment:
All functions declarations should clearly specify arguments types. What is the model_order type? Is it Int?

Reply:
I've been attempting to follow this recommendation from the Julia style guide: https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-writing-overly-specific-types. In cases where only a specific type is possible, I will add the type declaration.